docs.kosli.com/content/tutorials/unauthorized_iac_changes.md

```diff
@@ -0,0 +1,166 @@
+For simplicity, we will not define any requirements for this process by using `--use-empty-template`
+
+```shell {.command}
+$ kosli create flow tf-tutorial --use-empty-template
+```
+
+## Making and tracking an authorized change
+
+{{<hint info>}}
+In production, an authorized change will normally go though CI.
+In this tutorial, however, we run the commands that you would otherwise do in CI locally for simplicity.
+{{</hint>}}
+
+Let's create a trail to represent a single instance of making an authorized change. We will call it `authorized-1`.
+
+```shell {.command}
+$ kosli begin trail authorized-1 --flow tf-tutorial
+```
+Next, we can scan our terraform config scripts for security issues. We capture the SARIF output from the scan and attest it to Kosli.
+
+```shell {.command}
+$ snyk iac test main.tf --sarif-file-output=sarif.json
+$ kosli attest snyk --name security --flow tf-tutorial --trail authorized-1 --scan-results sarif.json
+```
+
+We are now ready to run terraform. We create a plan and save it to a file. Then attest the plan file to Kosli to build a historical audit log. 
+
+```shell {.command}
+$ terraform init
+$ terraform plan -out=tf.plan
+$ kosli attest generic --name tf-plan --flow tf-tutorial --trail authorized-1 --attachments tf.plan
+```
+
+Finally, we apply the terraform plan, and attest the produced terraform state file as an artifact.
+This will calculate a SHA256 fingerprint for the state file based on its contents. The fingerprint will later be used to determine if a change is 
+authorized or not.
+
+{{<hint info>}}
+In this tutorial, we use a simple setup where the terraform state file is stored locally.
+In production cases, however, the state file would be stored in some cloud storage (e.g. AWS S3)
+
+Note that we set both `--build-url` and `--commit-url` to fake URLs. These are normally defaulted in CI.
+{{</hint>}}
+
+```shell {.command}
+$ terraform apply -auto-approve tf.plan
+$ kosli attest artifact terraform.tfstate --name state-file --artifact-type file --flow tf-tutorial --trail authorized-1 \
+   --build-url https://example.com --commit-url https://example.com --commit HEAD
+```
+
+## Monitoring the state file
+
+Every time a change to the infrastructure happens via Terraform, the state file content would be changed. 
+To detect when an **unauthorized** change happens, we can monitor the state file for changes and record those changes in
+a Kosli environment.
+
+Let's start by creating an environment of type `server`. 
+
+```shell {.command}
+$ kosli create env terraform-state --type server
+```
+
+We can report the state file to the environment we created:
+
+{{<hint info>}}
+In this tutorial, we run the environment reporting manually. 
+In production, you would configure the environment reporting to run periodically or on changes.
+{{</hint>}}
+
+```shell {.command}
+$ kosli snapshot server terraform-state -p terraform.tfstate
```

Comment:
We seem to be using --long-flags in other command
so maybe replace `-p` with `--paths`

Reply:
done